Automattic/wp-calypso
# Add hosting config page

This PR moves the hosting config from the hosting wp-admin menu to the Global site view (GSV).

Ref: https://github.com/Automattic/dotcom-forge/issues/6255

### Nav Uni - Hosting config (where it is now)

<img width="903" alt="Screenshot 2024-04-22 at 11 54 33" src="https://github.com/Automattic/wp-calypso/assets/5560595/b7981448-cb58-4843-b4f6-1a66b1d1563b">

### GSV - Hosting Config

<img width="1283" alt="Screenshot 2024-04-22 at 11 52 59" src="https://github.com/Automattic/wp-calypso/assets/5560595/c706a19f-c3cf-4ebe-a0f8-ff6fc39722dd">

This page will eventually be a tab in the sites dashboard.

### Testing Instructions

Confirm hosting config loads from a URL similar to `/hosting-config/{atomic-site}`
Confirm doesn't affect the current sidebars when feature flag is disabled
To enable feature flag include `?flags=layout/dotcom-nav-redesign-v2`
To disable feature flag include `?flags=-layout/dotcom-nav-redesign-v2`



## TODO

- [x] Add handler for hosting config pages
- [x] Merge https://github.com/Automattic/wp-calypso/pull/89771 and rebase this PR
- [x] Update `getShouldShowGlobalSidebar` to support showing global sidebar when feature flag is enabled
- [ ] Possibly refactor `client/hosting-overview` to be some something like `client/hosting-tools` or just `client/hosting`
